crates/hir-def/src/hir/type_ref.rs

```diff
@@ -380,73 +381,84 @@ impl TypeBound {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum ConstRefOrPath {
-    Scalar(ConstRef),
+pub enum ConstRef {
+    Scalar(ConcreteConstRef),
     Path(Name),
+    Complex(InFile<FileAstId<ast::Expr>>),
```

Comment:
```suggestion
    Complex(AstId<ast::Expr>),
```
we have a convenient alias for that